platforms/x86-64/core/src/lib.rs

```diff
@@ -0,0 +1,110 @@
+#![no_std]
+extern crate alloc;
+
+use core::time::Duration;
+use hal_core::{boot::BootInfo, PAddr};
+use hal_x86_64::cpu::local::GsLocalData;
+pub use hal_x86_64::{
+    cpu::{local::LocalKey, wait_for_interrupt},
+    mm,
+};
+use kernel::{mnemos_alloc::containers::Box, Kernel, KernelSettings};
+
+pub mod acpi;
+pub mod interrupt;
+
+pub fn init(bootinfo: &impl BootInfo, rsdp_addr: Option<PAddr>) -> &'static Kernel {
+    // TODO: init early tracing?
```

Comment:
would be nice to have framebuffer and/or UART tracing before the runtime comes up...